tensorflow/compiler/tf2xla/kernels/BUILD

```diff
@@ -150,7 +150,7 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         "//tensorflow/compiler/tf2xla:xla_local_runtime_context",
-        "//tensorflow/core:framework_lite",
+        "//tensorflow/core:framework",
```

Comment:
This change in deps from "framework_lite" to "framework", and others like it in this PR, are problematic.

The issue is that we use gather_op_kernel_float_int32 (and other similar cc_libraries) in XLA ahead-of-time builds, via tfcompile.  We need to keep runtime dependencies at a minimum for these libraries, since some of our users on mobile (i.e. --config=android_arm) are relying on small generated codesize.

Furthermore I'm not sure whether the full "//tensorflow/core:framework" actually builds on --config=android_arm; I remember encountering problems here before.

Reply:
We have to dependent some framework headers if use multiple thread. Actually, there are some ops' dependencies include "framework". 
I added a new rule which only contains the headers `Shard` depend on in `core/BUILD` to minize the library.